### PR TITLE
feat(create-branch): handle tokens for private scoped packages

### PR DESCRIPTION
**Pull request by hulkoba**

### CODE REVIEW

File: lib/create-branch.js [hunk 1]
```diff
@@ -106,7 +106,31 @@ module.exports = async (
...
+            npm: ${token},
+            github: ${token}
+          }
+        }
+        */
+        let repositoryTokens = null
+        try {
+          repositoryTokens = await tokens.get(installationId)
+
+          const datetime = new Date().toISOString().substr(0, 19).replace('T', '_')
```
Comment (hulkoba) [inline lib/create-branch.js]: The token_audit id look like
`123:one-lockfile:2018-11-16_14:46:00:read`
I'm not sure if the datetime is nice. Julia wants everything to be dashes.

Comment (hulkoba) [reply lib/create-branch.js]: we’re all in favour of the format '20181116144600'

File: lib/create-branch.js [hunk 2]
```diff
@@ -106,7 +106,32 @@ module.exports = async (
 
       const isNpm = lockfilePath.includes('package-lock.json')
       try {
-        const { ok, contents } = await getNewLockfile(commit.content, oldLockfileContent, isNpm)
+        const { tokens, tokens_audit } = await dbs() // eslint-disable-line
```
[the remark this answers is not in the document]
Comment (hulkoba) [reply lib/create-branch.js]: @espy it is `token-audits` now :)

File: lib/create-branch.js [hunk 3]
```diff
@@ -106,7 +106,32 @@ module.exports = async (
...
+          const datetime = new Date().toISOString().substr(0, 19).replace(/[^0-9]/g, '')
+
+          // write audit log entry to 'tokens-audit' db
+          // log entry type: 'read'
+          await tokenAudits.put({
+            _id: `${installationId}:${repoDoc._id}:${datetime}:read`,
+            keys: Object.keys(repositoryTokens.tokens[repoDoc._id])
+          })
+        } catch (e) {}
+        const { ok, contents } = await getNewLockfile({ packageJson: commit.content, lock: oldLockfileContent, isNpm, repositoryTokens: JSON.stringify(repositoryTokens) })
```
Comment (Realtin) [inline lib/create-branch.js]: repositoryTokens should also be `repositoryTokens.tokens[repoDoc._id]` so we only get the tokens we need

Comment (janl) [reply lib/create-branch.js]: +1

**Pull request merged**